bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -0,0 +1,290 @@
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writefunc);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &s);
+        if(tunnelUsed){
+            curl_easy_setopt(curl, CURLOPT_PROXY, tunnel);
+        }
+        curl_easy_setopt(curl, CURLOPT_POSTFIELDS, data);
+        /* Perform the request, res will get the return code */
+        res = curl_easy_perform(curl);
+        /* Check for errors */
+        if(res != CURLE_OK){
+            fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
+            s.ptr = "FAILED";
+            s.len = strlen(s.ptr);
+            return s;
+        }
+
+        /* always cleanup */
+        curl_easy_cleanup(curl);
+    }
+    curl_global_cleanup();
+    return s;
+}
+
+char** getIpAddresses(int maxSize, int *addrCount){
+    char** IPs = malloc(maxSize * sizeof(char*));
+    int i = 0;
+
+    struct ifaddrs * ifAddrStruct=NULL;
+    struct ifaddrs * ifa=NULL;
+    void * tmpAddrPtr=NULL;
+    getifaddrs(&ifAddrStruct);
+
+    for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next) {
+        if (!ifa->ifa_addr || i >= maxSize) {
+            continue;
+        }
+        if (ifa->ifa_addr->sa_family == AF_INET) { // check it is IP4
+            // is a valid IP4 Address
+            tmpAddrPtr=&((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+            IPs[i] = malloc(INET_ADDRSTRLEN);
+            inet_ntop(AF_INET, tmpAddrPtr, IPs[i], INET_ADDRSTRLEN);
+            i = i+1;
+        }
+    }
+    if (ifAddrStruct!=NULL) freeifaddrs(ifAddrStruct);
+    *addrCount = i;
+    return IPs;
+}
+
+int ping_island(int argc, char * argv[])
+{
+    // Get system info
+    struct utsname systemInfo;
+
+    // Get glibc version
+    char* glibcVersion = executeCommand("ldd --version");
+    printf("glibc: %s\n", glibcVersion);
+
+    char* osVersion = executeCommand("cat /etc/os-release");
+    if(!strcmp(osVersion, "")){
+        osVersion = executeCommand("cat /etc/system-release");
+    }
+    printf("Os version: %s \n", osVersion);
+
+    // Get all machine IP's
+    const int maxIPs = 20;
+    int addrCount = 0;
+    char** IPs = getIpAddresses(maxIPs, &addrCount);
+    char* IPstring = concatenate(addrCount, IPs, "\", \"");
+
+    // Get hostname
+    char hostname[HOST_NAME_MAX + 1];
+    gethostname(hostname, HOST_NAME_MAX + 1);
+    printf("Hostname: %s\n", hostname);
+
+    // Form request struct
+    struct requestData reqData;
+    reqData.osVersion = osVersion;
+    reqData.hostname = hostname;
+    reqData.IPstring = IPstring;
+    reqData.tunnelUsed = false;
+    reqData.glibcVersion = glibcVersion;
+
+
+    int i, tunnel_i=0, server_i=0;
+    char t_flag[] = "-t";
+    char s_flag[] = "-s";
+    for(i=1;i<argc;i++)
+    {
+        if(strcmp(argv[i],t_flag) == 0){
+            tunnel_i = i+1;
+        } else if(strcmp(argv[i],s_flag) == 0){
+            server_i = i+1;
+        }
+    }
+
+    char* server = argv[server_i];
```

Comment:
https://www.gnu.org/software/libc/manual/html_node/Example-of-Getopt.html#Example-of-Getopt

Reply:
I don't see how that's better